### PR TITLE
Make axes gui compatible with non-linear-axis

### PR DESCRIPTION
**Pull request by jlaehne**
### Description of the change

Gui for axes failed when `offset` or `scale` parameters were missing. Added checks for existance of these attributes to make the gui compatible with [hyperspy-PR#1927](https://github.com/hyperspy/hyperspy/pull/1927) now continued in the [non-uniform-axes branch](https://github.com/hyperspy/hyperspy/tree/non_uniform_axes).

### Progress of the PR:

- [X] Change implemented and tested
- [X] Ready for review

### CODE REVIEW

File: hyperspy_gui_ipywidgets/axes.py [hunk 1]
```diff
@@ -94,15 +94,23 @@ def _get_axis_widgets(obj):
     link((obj, "units"), (units, "value"))
     wd["units"] = units
 
-    scale = ipywidgets.FloatText()
-    widgets.append(labelme("Scale", scale))
-    link((obj, "scale"), (scale, "value"))
-    wd["scale"] = scale
+    if "scale" in obj.__dict__.keys():
```
Comment (francisco-dlp) [inline hyperspy_gui_ipywidgets/axes.py]: What about using ``hasattr(obj, "scale")`` instead?

File: hyperspy_gui_ipywidgets/axes.py [hunk 2]
```diff
@@ -94,15 +94,23 @@ def _get_axis_widgets(obj):
...
+        scale = ipywidgets.FloatText()
+        widgets.append(labelme("Scale", scale))
+        link((obj, "scale"), (scale, "value"))
+        wd["scale"] = scale
 
-    offset = ipywidgets.FloatText()
-    widgets.append(labelme("Offset", offset))
-    link((obj, "offset"), (offset, "value"))
-    wd["offset"] = offset
+    if "offset" in obj.__dict__.keys():
```
Comment (francisco-dlp) [inline hyperspy_gui_ipywidgets/axes.py]: as above

File: hyperspy_gui_ipywidgets/axes.py (hunk 1 above)
Comment (jlaehne) [reply hyperspy_gui_ipywidgets/axes.py]: Good, changed 3 times

**Comment by jlaehne** [comment]
Added test, which for the moment fails (as long as non-linear-axis branch is not merged in HyperSpy).

**Pull request merged**